File: boj/DFS와BFS/1260_dfs와bfs.py
from collections import deque

def dfs(graph, v, visited):
	visited[v] = True
	print(v, end=' ')
	
	for i in graph[v]:
		if not visited[i]:
			dfs(graph, i, visited)

def bfs(graph, v, visited):
	queue = deque([v]) # 큐 자료구조 구현 위해 덱 사용 -> 빨라빨라
	visited[v] = True
	
	while queue:
		now = queue.popleft()
		print(now, end=' ')
		for i in graph[now]:
			if not visited[i]:
				visited[i] = True
				queue.append(i)
				
n, m, v = map(int, input().rstrip().split())

graph = [[] for _ in range(n+1)]
visited = [False] * (n+1)

for _ in range(m):
	a, b = map(int, input().rstrip().split())
	graph[a].append(b)
	graph[b].append(a)

for g in graph: # 낮은 숫자부터 비교 -> 오름차순 정렬
	g.sort()

dfs(graph, v, visited)

visited = [False] * (n+1) # 방문 배열 다시 초기화
print()

bfs(graph, v, visited)

File: boj/DFS와BFS/1260_DFS와BFS.cpp
// dfs 와 bfs 기본 알고리즘 구조 파악하기 좋은 문제
// 차이 및 사용법 익히자
#include <iostream>
#include <vector>
#include <algorithm>
#include <queue>
using namespace std;
void dfs(int s,vector<int> graph[],bool visited[]){
	visited[s]=true;
	cout<<s<<" ";
	
	for(int i=0;i<graph[s].size();i++){
		int next=graph[s][i];
		
		if(!visited[next])
			dfs(next,graph,visited);
	}
}
void bfs(int s,vector<int> graph[],bool visited[]){
	queue<int> q;
	
	q.push(s);
	visited[s]=true;
	
	while(!q.empty()){
		int start=q.front();
		q.pop();
		cout<<start<<" ";
		
		for(int i=0;i<graph[start].size();i++){
			int next=graph[start][i];
			
			if(!visited[next]){
				q.push(next);
				visited[next]=true;
			}
		}
	}
}
int main() {
	int n,m,v;
	cin>>n>>m>>v;
	
	vector<int> graph[n+1];
	bool visited[n+1];
	fill(visited,visited+(n+1),false);//bool false로 값 초기화
	
	for(int i=0;i<m;i++){
		int u,v;
		cin>>u>>v;
		
		graph[u].push_back(v);
		graph[v].push_back(u);
	}
	
	for(int i=0;i<n;i++)
		sort(graph[i+1].begin(),graph[i+1].end());
	
	dfs(v,graph,visited);
	cout<<"\n";
	fill(visited,visited+(n+1),false);
	bfs(v,graph,visited);
	cout<<"\n";
	
	return 0;
}